apps/bfd-pipeline/bfd-pipeline-ccw-rif/src/test/java/gov/cms/bfd/pipeline/ccw/rif/extract/s3/DataSetManifestTest.java

```diff
@@ -30,22 +34,78 @@ public void jaxbUnmarshallingForSampleA() throws JAXBException {
     InputStream manifestStream =
         Thread.currentThread().getContextClassLoader().getResourceAsStream("manifest-sample-a.xml");
 
-    JAXBContext jaxbContext = JAXBContext.newInstance(DataSetManifest.class);
-    Unmarshaller jaxbUnmarshaller = jaxbContext.createUnmarshaller();
-
-    DataSetManifest manifest = (DataSetManifest) jaxbUnmarshaller.unmarshal(manifestStream);
+    DataSetManifest manifest = DataSetManifestFactory.newInstance().parseManifest(manifestStream);
 
     assertNotNull(manifest);
     assertEquals(
         1994,
         LocalDateTime.ofInstant(manifest.getTimestamp(), ZoneId.systemDefault())
             .get(ChronoField.YEAR));
     assertEquals(1, manifest.getSequenceId());
+    // xml did not contain the syntheticData attribute; therefore the isSyntheticData should be
+    // false.
+    assertEquals(false, manifest.isSyntheticData());
     assertEquals(2, manifest.getEntries().size());
     assertEquals("sample-a-beneficiaries.txt", manifest.getEntries().get(0).getName());
     assertEquals(RifFileType.BENEFICIARY, manifest.getEntries().get(0).getType());
   }
 
+  /**
+   * Verifies that {@link DataSetManifest} cannot be unmarshalled, as expected. The invalid-sample-a
+   * XML defines the syntheticData with a value of "junk"; this should trigger a {@link
+   * SAXException} which will result in a linked exception within {@link JAXBException}, {@link
+   * UnmarshalException}.
+   *
+   * <p>The error messsage should identify the XML attribute in error and the value that it
+   * attempted to use.
+   *
+   * <p>JAXBException (indicates test success)
+   */
+  @Test
+  public void jaxbUnmarshallingForInvalidSampleA() {
+    JAXBException thrown =
+        assertThrows(
+            JAXBException.class,
+            () -> {
+              InputStream manifestStream =
+                  Thread.currentThread()
+                      .getContextClassLoader()
+                      .getResourceAsStream("manifest-invalid-sample-a.xml");
+
+              DataSetManifestFactory.newInstance().parseManifest(manifestStream);
+            },
+            "SAXParseException message was expected");
+
+    UnmarshalException unmarshalException = (UnmarshalException) thrown.getLinkedException();
+    assertNotNull(unmarshalException);
```

Comment:
This would be a good place for `assertInstanceOf`:

```
assertNotNull(unmarshalException);
UnmarshalException unmarshalException = assertInstanceOf(UnmarshalException.class, thrown.getLinkedException());
```

That way if the exception is not an instance of `UnmarshalException` you'd get a meaningful error message instead of a `ClassCastException`.

Reply:
Brian, thx, I glossed over that and your idea is perfect. Implemented with the assertInstanceof